### PR TITLE
LUCENE-10000: Make MultiCollectorManager consistent with MultiCollector

### PR DESCRIPTION
**Pull request by gsmiller**
# Description

MultiCollectorManager is now consistent with MultiCollector with respect to early termination, min score setting and score caching.

# Solution

MultiCollectorManager delegates to MultiCollector instead of creating its own solution for wrapping/managing multiple Collectors.

# Tests

Created unit tests for MultiCollectorManager (none previously existed). Tests cover various scenarios that are similarly covered in MultiCollector.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/lucene/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Lucene maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -236,6 +236,9 @@ Improvements
 * LUCENE-9535: Improve DocumentsWriterPerThreadPool to prefer larger instances.
   (Adrien Grand)
 
+* LUCENE-10000: MultiCollectorManager now has parity with MultiCollector with respect to how it
+  handles CollectionTerminationException and minCompetitiveScore calls. (Greg Miller)
```
Comment (jpountz) [inline lucene/CHANGES.txt:240]:
```suggestion
  handles CollectionTerminationException and setMinCompetitiveScore calls. (Greg Miller)
```

File: lucene/core/src/java/org/apache/lucene/search/MultiCollector.java [hunk 2]
```diff
@@ -83,7 +83,7 @@ public static Collector wrap(Iterable<? extends Collector> collectors) {
   }
 
   private final boolean cacheScores;
-  private final Collector[] collectors;
+  final Collector[] collectors;
```
Comment (jpountz) [inline lucene/core/src/java/org/apache/lucene/search/MultiCollector.java:86]: nit: can you keep it private and add a pkg-private getter instead?

File: lucene/core/src/test/org/apache/lucene/search/TestMultiCollectorManager.java [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        expected.add(random().nextInt());
+      }
+      List<Integer> expectedEven =
+          expected.stream().filter(evenPredicate).collect(Collectors.toList());
+      List<Integer> expectedOdd =
+          expected.stream().filter(oddPredicate).collect(Collectors.toList());
+
+      // Test only wrapping one of the collector managers:
+      MultiCollectorManager mcm = new MultiCollectorManager(cm1);
+      Object[] results = (Object[]) collectAll(ctx, expected, mcm);
```
Comment (jpountz) [inline lucene/core/src/test/org/apache/lucene/search/TestMultiCollectorManager.java:63]: let's have `expected` sorted and only containing unique doc IDs to be more realistic?

File: lucene/core/src/java/org/apache/lucene/search/MultiCollector.java (hunk 2 above)
Comment (gsmiller) [reply lucene/core/src/java/org/apache/lucene/search/MultiCollector.java:86]: Sure!

File: lucene/core/src/test/org/apache/lucene/search/TestMultiCollectorManager.java (hunk 3 above)
Comment (gsmiller) [reply lucene/core/src/test/org/apache/lucene/search/TestMultiCollectorManager.java:63]: Good idea. Will do.

**Review by gsmiller — COMMENTED** [review]
Thanks for the feedback @jpountz

**Review by jpountz — APPROVED** [review]
Thanks for fixing this!

**Comment by gsmiller** [comment]
> Thanks for fixing this!

Sure. Thanks for the feedback!

**Pull request merged**